### PR TITLE
Update Source packages to use React 18

### PR DESCRIPTION
**Pull request by joecowton1**
## What are you changing?

- Upgrading React to version 18 across all packages in CSNX
- Upgrading versions of @guardian packages to latest versions

DO NOT RELEASE UNTIL W/0 24th APRIL

### CODE REVIEW

**Review by mxdvl — COMMENTED** [review]
Small inconsistencies — which can be caught by the package linter, if it ever reaches a production stage.

File: libs/@guardian/atoms-rendering/package.json [hunk 1]
```diff
@@ -38,10 +38,10 @@
 		"@guardian/consent-management-platform": "^11.0.0",
 		"@guardian/libs": "^14.0.0",
 		"@guardian/source-foundations": "^10.0.0",
-		"@guardian/source-react-components": "^12.0.0",
+		"@guardian/source-react-components": "^13.0.0",
 		"lodash.debounce": "^4.0.6",
-		"react": "^17.0.1",
-		"react-dom": "^17.0.1",
+		"react": "^18.0.0",
+		"react-dom": "^18.0.0",
```
[the remark this answers is not in the document]
Comment (joecowton1) [reply libs/@guardian/atoms-rendering/package.json:44]: This is the approach we've used throughout CSNX and in Source previously - is this something you disagree with?

File: libs/@guardian/ab-react/package.json [hunk 2]
```diff
@@ -9,16 +9,16 @@
 		"@emotion/react": "11.0.0",
 		"@guardian/ab-core": "4.0.0",
 		"@testing-library/react": "11.2.2",
-		"@types/react": "~17.0",
-		"react": "17.0.2",
-		"react-dom": "17.0.2",
+		"@types/react": "18.0.28",
```
[the remark this answers is not in the document]
Comment (joecowton1) [reply libs/@guardian/ab-react/package.json:12]: There isn't a @types/react `18.2.x`, but there are a lot of fixes in React `18.2.x` - are you suggesting we don't use React `18.2.x` until there's a matching types package?

[the remark this answers is not in the document]
Comment (mxdvl) [reply libs/@guardian/ab-react/package.json]: That would be the safest option, but this should not block our work either.

We should try and help with [raising a new `@types` package that matches `18.2.x`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts).

File: libs/@guardian/atoms-rendering/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (mxdvl) [reply libs/@guardian/atoms-rendering/package.json:44]: Are you sure? I cannot find an example and we specifically have a recommendation against it.

> [Pin the lowest possible version of peerDependencies in your package's devDependencies](https://github.com/guardian/recommendations/blob/main/npm-packages.md#pin-the-lowest-possible-version-of-peerdependencies-in-your-packages-devdependencies)

[the remark this answers is not in the document]
Comment (joecowton1) [reply libs/@guardian/atoms-rendering/package.json]: Yeah point taken, updated now 👍

**Pull request merged**